### PR TITLE
959 add information about error handling to the api docs

### PR DESCRIPTION
**Pull request by RobertM905**
### Context

Vendors have requested documentation to clarify potential error responses from the api.

### Changes proposed in this pull request

- Add an Errors page to the documentation to outline the basic structure of errors and some examples of potential errors while using the api.

### Guidance to review

Check the Errors page to ensure that the documentation makes sense.

### Release notes

- [x ] [If needed](/README.md#release-notes), the [release notes](/source/release-notes.html.md) have been updated

### Link to Trello card

[959 - add information about error handling to the api docs](https://trello.com/c/HrYp2uiD/959-add-information-about-error-handling-to-the-api-docs)

### CODE REVIEW

File: lib/application_json.rb [hunk 1]
```diff
@@ -402,6 +402,31 @@ def rejection_attributes
...
+    [
+      {
+        name: 'errors',
+        type: 'array',
+        description: 'Array containing multiple errors'
+      },
+      {
+        name: 'status',
+        type: 'string',
+        description: 'http status code for the error'
```
Comment (alexdesi) [inline lib/application_json.rb]: should Http be in capital?

File: source/errors.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+{
+  "errors": [
+    {
+      "status": "422",
+      "title":  "Invalid Attribute",
+      "detail": "Volume does not, in fact, go to 11."
+    }
+  ]
+}
+```
```
Comment (alexdesi) [inline source/errors.html.md.erb]: In this way is not easy to identify the field name (title for example).
what about this something like:
```
{
  "errors": [
    {
      "status": "422",
      "field": "title",
      "error_message":  "Invalid Attribute",
      "detail": "Volume does not, in fact, go to 11."
    }
  ]
}
```

Comment (alexdesi) [reply source/errors.html.md.erb]: the format supported by OpenAPI should be something like:
```
  schemas:
    # Schema for error response body
    Error:
      type: object
      properties:
        code:
          type: string
        message:
          type: string
      required:
        - code
        - message
```
see https://swagger.io/docs/specification/describing-responses/

**Comment by RobertM905** [comment]
Closing this PR as we will be adding error responses to the work included in the move to OpenApi.
See: https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training-tech-docs/pull/64

**Pull request closed without merging**